chia/data_layer/dl_wallet_store.py

```diff
@@ -108,11 +108,29 @@ async def add_singleton_record(self, record: SingletonRecord, in_transaction: bo
             if not in_transaction:
                 self.db_wrapper.lock.release()
 
-    async def get_all_singletons_for_launcher(self, launcher_id: bytes32) -> List[SingletonRecord]:
+    async def get_all_singletons_for_launcher(
+        self,
+        launcher_id: bytes32,
+        min_generation: Optional[uint32] = None,
+        max_generation: Optional[uint32] = None,
+        num_results: Optional[uint32] = None,
```

Comment:
why do we need the ability  to limit number of results between generations ?

Reply:
This is just generally consistent with how we do paginations in other stores.

Reply:
As a general comment, not a change request here, I think an API with separate min/max and min/count requests might be good.  Then there doesn't have to be any prioritization of whether max or count wins or the associated documentation and test explosion.